Build error

On my CI I'm getting

```
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:45:0,
1086                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
1087                 from /home/travis/miniconda/include/xsimd/config/xsimd_include.hpp:19,
1088                 from /home/travis/miniconda/include/xsimd/types/xsimd_types_include.hpp:14,
1089                 from /home/travis/miniconda/include/xsimd/types/xsimd_traits.hpp:17,
1090                 from /home/travis/miniconda/include/xsimd/xsimd.hpp:16,
1091                 from /home/travis/miniconda/include/xtensor/xtensor_config.hpp:84,
1092                 from /home/travis/miniconda/include/xtensor/xbuffer_adaptor.hpp:21,
1093                 from /home/travis/miniconda/include/xtensor/xtensor.hpp:19,
1094                 from /home/travis/build/tdegeus/GooseEYE/include/GooseEYE/config.h:26,
1095                 from /home/travis/build/tdegeus/GooseEYE/include/GooseEYE/GooseEYE.h:10,
1096                 from /home/travis/build/tdegeus/GooseEYE/test/main.cpp:2:
1097/home/travis/miniconda/include/xsimd/types/xsimd_avx512_int64.hpp: In function ‘xsimd::batch<long int, 8ul> xsimd::operator<<(const xsimd::batch<long int, 8ul>&, int32_t)’:
1098/home/travis/miniconda/include/xsimd/types/xsimd_avx512_int64.hpp:412:16: error: the last argument must be an 8-bit immediate
1099         return _mm512_slli_epi64(lhs, rhs);
1100                ^
1101/home/travis/miniconda/include/xsimd/types/xsimd_avx512_int64.hpp: In function ‘xsimd::batch<long int, 8ul> xsimd::operator>>(const xsimd::batch<long int, 8ul>&, int32_t)’:
1102/home/travis/miniconda/include/xsimd/types/xsimd_avx512_int64.hpp:417:16: error: the last argument must be an 8-bit immediate
1103         return _mm512_srli_epi64(lhs, rhs);
1104                ^
```

FYI: https://travis-ci.org/tdegeus/GooseEYE/jobs/623781528

--- a/.travis.yml
+++ b/.travis.yml
@@ -40,8 +40,8 @@ matrix:
           sources:
             - ubuntu-toolchain-r-test
           packages:
-            - g++-6
-      env: COMPILER=gcc GCC=6 ENABLE_FALLBACK=1 ENABLE_XTL_COMPLEX=1
+            - g++-7
+      env: COMPILER=gcc GCC=7 ENABLE_FALLBACK=1 ENABLE_XTL_COMPLEX=1
     - os: linux
       sudo: required
       addons:
@@ -197,7 +197,7 @@ script:
         fi
       elif [[ "$AVX512" == 1 ]] ; then
         uname -m ;
-        ../intel-sde-downloader/sde-external-8.35.0-2019-03-11-lin/sde64 -cnl -- ./test_xsimd ;
+        ../intel-sde-downloader/sde-external-8.35.0-2019-03-11-lin/sde64 -skx -- ./test_xsimd ;
       else
         ./test_xsimd;
       fi

--- a/include/xsimd/types/xsimd_avx512_int32.hpp
+++ b/include/xsimd/types/xsimd_avx512_int32.hpp
@@ -354,12 +354,16 @@ namespace xsimd
 
     inline batch<int32_t, 16> operator<<(const batch<int32_t, 16>& lhs, int32_t rhs)
     {
-        return _mm512_slli_epi32(lhs, rhs);
+        // _mm512_slli_epi32 expects its last argument to be known at compile time,
+        // which cannot be guaranteed here.
+        return _mm512_sllv_epi32(lhs, batch<int32_t, 16>(rhs));
     }
 
     inline batch<int32_t, 16> operator>>(const batch<int32_t, 16>& lhs, int32_t rhs)
     {
-        return _mm512_srli_epi32(lhs, rhs);
+        // _mm512_srli_epi32 expects its last argument to be known at compile time,
+        // which cannot be guaranteed here.
+        return _mm512_srlv_epi32(lhs, batch<int32_t, 16>(rhs));
     }
 
     inline batch<int32_t, 16> operator<<(const batch<int32_t, 16>& lhs, const batch<int32_t, 16>& rhs)
@@ -374,12 +378,16 @@ namespace xsimd
 
     inline batch<uint32_t, 16> operator<<(const batch<uint32_t, 16>& lhs, int32_t rhs)
     {
-        return _mm512_slli_epi32(lhs, rhs);
+        // _mm512_slli_epi32 expects its last argument to be known at compile time,
+        // which cannot be guaranteed here.
+        return _mm512_sllv_epi32(lhs, batch<uint32_t, 16>(rhs));
     }
 
     inline batch<uint32_t, 16> operator>>(const batch<uint32_t, 16>& lhs, int32_t rhs)
     {
-        return _mm512_srli_epi32(lhs, rhs);
+        // _mm512_srli_epi32 expects its last argument to be known at compile time,
+        // which cannot be guaranteed here.
+        return _mm512_srlv_epi32(lhs, batch<int32_t, 16>(rhs));
     }
 
     inline batch<uint32_t, 16> operator<<(const batch<uint32_t, 16>& lhs, const batch<int32_t, 16>& rhs)

--- a/include/xsimd/types/xsimd_avx512_int64.hpp
+++ b/include/xsimd/types/xsimd_avx512_int64.hpp
@@ -409,12 +409,16 @@ namespace xsimd
 
     inline batch<int64_t, 8> operator<<(const batch<int64_t, 8>& lhs, int32_t rhs)
     {
-        return _mm512_slli_epi64(lhs, rhs);
+        // _mm512_slli_epi64 expects its last argument to be known at compile time,
+        // which cannot be guaranteed here.
+        return _mm512_sllv_epi64(lhs, batch<int64_t, 8>(rhs));
     }
 
     inline batch<int64_t, 8> operator>>(const batch<int64_t, 8>& lhs, int32_t rhs)
     {
-        return _mm512_srli_epi64(lhs, rhs);
+        // _mm512_srli_epi64 expects its last argument to be known at compile time,
+        // which cannot be guaranteed here.
+        return _mm512_srlv_epi64(lhs, batch<int64_t, 8>(rhs));
     }
 
     inline batch<int64_t, 8> operator<<(const batch<int64_t, 8>& lhs, const batch<int64_t, 8>& rhs)
@@ -429,12 +433,16 @@ namespace xsimd
 
     inline batch<uint64_t, 8> operator<<(const batch<uint64_t, 8>& lhs, int32_t rhs)
     {
-        return _mm512_slli_epi64(lhs, rhs);
+        // _mm512_slli_epi64 expects its last argument to be known at compile time,
+        // which cannot be guaranteed here.
+        return _mm512_sllv_epi64(lhs, batch<uint64_t, 8>(rhs));
     }
 
     inline batch<uint64_t, 8> operator>>(const batch<uint64_t, 8>& lhs, int32_t rhs)
     {
-        return _mm512_srli_epi64(lhs, rhs);
+        // _mm512_srli_epi64 expects its last argument to be known at compile time,
+        // which cannot be guaranteed here.
+        return _mm512_srlv_epi64(lhs, batch<uint64_t, 8>(rhs));
     }
 
     inline batch<uint64_t, 8> operator<<(const batch<uint64_t, 8>& lhs, const batch<int64_t, 8>& rhs)

--- a/test/xsimd_cpower_test.cpp
+++ b/test/xsimd_cpower_test.cpp
@@ -112,12 +112,14 @@ TEST(xsimd, avx512_complex_double_power)
 }
 
 #if XSIMD_ENABLE_XTL_COMPLEX
+#if DEBUG_FLOAT_ACCURACY
 TEST(xsimd, avx512_xtl_xcomplex_float_power)
 {
     std::ofstream out("log/avx512_xtl_xcomplex_float_power.log", std::ios_base::out);
     bool res = xsimd::test_cpower<xtl::xcomplex<float>, 16, 64>(out, "avx512 xtl xcomplex float");
     EXPECT_TRUE(res);
 }
+#endif
 
 TEST(xsimd, avx512_xtl_xcomplex_double_power)
 {

